not supported ATM to reference additional remotes [network.py:as_git_url:782] (NotImplementedError)

#### What is the problem?
`datalad install` does not work
#### What steps will reproduce the problem?
```
me@christop ~ $ datalad search NSF
No DataLad dataset found at current location
Would you like to install the DataLad superdataset at 'C:\\Users\\me\\datalad'? (choices: yes, no): ERROR:
Interrupted by user while doing magic: KeyboardInterrupt() [getpass.py:win_getpass:110]
me@christop ~ $ cd /c/stanford_gdrive/data/
me@christop /c/stanford_gdrive/data $ datalad search NSF
No DataLad dataset found at current location
Would you like to install the DataLad superdataset at 'C:\\Users\\me\\datalad'? (choices: yes, no):
[INFO] Cloning http://datasets.datalad.org/ to 'C:\Users\me\datalad'
From now on you can refer to this dataset using the label '///'
[INFO] Performing search using DataLad superdataset 'C:\\Users\\me\\datalad'
[INFO] Reached the limit of 20 top matches, there could be more which were not reported.
search(ok): C:\Users\me\datalad\crcns\cai-2 (dataset)
search(ok): C:\Users\me\datalad\crcns\hc-11 (dataset)
search(ok): C:\Users\me\datalad\hbnssi (dataset)
search(ok): C:\Users\me\datalad\labs\haxby\attention (dataset)
search(ok): C:\Users\me\datalad\labs\haxby\life (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000001 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000002 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000003 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000005 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000006 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000007 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000008 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000011 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000017 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000030 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000051 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000052 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000101 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000102 (dataset)
search(ok): C:\Users\me\datalad\openfmri\ds000105 (dataset)
action summary:
  search (ok: 20)
me@christop /c/stanford_gdrive/data $ datalad install ///openfmri/ds000002
[ERROR] not supported ATM to reference additional remotes [network.py:as_git_url:782] (NotImplementedError)
```
#### What version of DataLad are you using (run `datalad --version`)? On what operating system (consider running `datalad wtf`)?
```
me@christop /c/stanford_gdrive/data $ datalad wtf
DataLad
=======
Version     : 0.10.0
Full version: 0.10.0

System
======
OS          : nt Windows 10 10.0.17134
Distribution:   10/10.0.17134/SP0/Multiprocessor Free

Locale/Encoding
===============
default        : utf-8
filesystem     : utf-8
locale.prefered: cp1252

Environment
===========
LANG='en_US.UTF-8'
PATH='c:\\users\\me\\miniconda3\\Library\\bin;C:\\Users\\me\\bin;C:\\Program Files\\Git\\mingw64\\bin;C:\\Program Files\\Git\\usr\\local\\bin;C:\\Program Files\\Git\\usr\\bin;C:\\Program Files\\Git\\usr\\bin;C:\\Program Files\\Git\\cmd;C:\\Program Files\\ConEmu\\ConEmu\\Scripts;C:\\Program Files\\ConEmu;C:\\Program Files\\ConEmu\\ConEmu;C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v9.2\\bin;C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v9.2\\libnvvp;C:\\Program Files (x86)\\Common Files\\Oracle\\Java\\javapath;C:\\Program Files\\Docker\\Docker\\Resources\\bin;C:\\Program Files (x86)\\Intel\\Intel(R) Management Engine Components\\iCLS;C:\\Program Files\\Intel\\Intel(R) Management Engine Components\\iCLS;C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System32\\Wbem;C:\\Windows\\System32\\WindowsPowerShell\\v1.0;C:\\Program Files (x86)\\Intel\\Intel(R) Management Engine Components\\DAL;C:\\Program Files\\Intel\\Intel(R) Management Engine Components\\DAL;C:\\Program Files (x86)\\Intel\\Intel(R) Management Engine Components\\IPT;C:\\Program Files\\Intel\\Intel(R) Management Engine Components\\IPT;C:\\Program Files\\Intel\\WiFi\\bin;C:\\Program Files\\Common Files\\Intel\\WirelessCommon;C:\\Program Files (x86)\\NVIDIA Corporation\\PhysX\\Common;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0;C:\\WINDOWS\\System32\\OpenSSH;C:\\ProgramData\\chocolatey\\bin;C:\\Program Files\\Git\\cmd;C:\\Program Files\\PuTTY;C:\\Program Files\\nodejs;C:\\Users\\me\\Miniconda3;C:\\Users\\me\\Miniconda3\\Library\\mingw-w64\\bin;C:\\Users\\me\\Miniconda3\\Library\\usr\\bin;C:\\Users\\me\\Miniconda3\\Library\\bin;C:\\Users\\me\\Miniconda3\\Scripts;C:\\Users\\me\\AppData\\Local\\Microsoft\\WindowsApps;C:\\Program Files\\Intel\\WiFi\\bin;C:\\Program Files\\Common Files\\Intel\\WirelessCommon;C:\\Users\\me\\AppData\\Local\\atom\\bin;C:\\Users\\me\\AppData\\Roaming\\npm;C:\\Program Files\\Git\\usr\\bin\\vendor_perl;C:\\Program Files\\Git\\usr\\bin\\core_perl'
PYTHONIOENCODING='UTF-8'

Externals
=========
appdirs=1.4.3
boto=2.48.0
cmd:annex=6.20180525-g940444994
cmd:git=2.17.1.windows.2
cmd:system-git=2.17.1.windows.2
cmd:system-ssh=7.7p1
git=2.1.10
gitdb=2.0.3
humanize=0.5.1
iso8601=0.1.12
msgpack=0.5.6
requests=2.18.4
six=1.11.0
tqdm=4.23.4
wrapt=1.10.11

Installed extensions
====================


Known metadata extractors
=========================
annex: OK
audio: BROKEN (No module named 'mutagen' [audio.py:<module>:17])
datacite: OK
datalad_core: OK
datalad_rfc822: OK
exif: BROKEN (No module named 'exifread' [exif.py:<module>:16])
frictionless_datapackage: OK
image: OK
xmp: BROKEN (No module named 'libxmp' [xmp.py:<module>:20])

Configuration
=============
<Sensitive data, use -s=some or -s=all to show>
```
#### Is there anything else that would be useful to know in this context?
I'm using 64bit installation of Git which is theoretically discouraged (at least it used to be).

#### Have you had any success using DataLad before? (to assess your expertise/prior luck.  We would welcome your testimonial additions to https://github.com/datalad/datalad/wiki/Testimonials as well)
You and me go way back.

--- a/datalad/__init__.py
+++ b/datalad/__init__.py
@@ -16,29 +16,9 @@ if _seed:
     import random
     random.seed(_seed)
 
-# Other imports are interspersed with lgr.debug to ease troubleshooting startup
-# delays etc.
-
-# If there is a bundled git, make sure GitPython uses it too:
-from datalad.cmd import GitRunner
-GitRunner._check_git_path()
-if GitRunner._GIT_PATH:
-    import os
-    os.environ['GIT_PYTHON_GIT_EXECUTABLE'] = \
-        os.path.join(GitRunner._GIT_PATH, 'git')
-
-from .config import ConfigManager
-cfg = ConfigManager()
-
-from .log import lgr
 import atexit
-from datalad.utils import on_windows, get_encoding_info, get_envvars_info
-
-lgr.log(5, "Instantiating ssh manager")
-from .support.sshconnector import SSHManager
-ssh_manager = SSHManager()
-atexit.register(ssh_manager.close, allow_fail=False)
-
+# Colorama (for Windows terminal colors) must be imported before we use/bind
+# any sys.stdout
 try:
     # this will fix the rendering of ANSI escape sequences
     # for colored terminal output on windows
@@ -54,6 +34,28 @@ except ImportError as e:
             "'colorama' Python module missing, terminal output may look garbled [%s]",
             exc_str(e))
     pass
+
+# Other imports are interspersed with lgr.debug to ease troubleshooting startup
+# delays etc.
+
+# If there is a bundled git, make sure GitPython uses it too:
+from datalad.cmd import GitRunner
+GitRunner._check_git_path()
+if GitRunner._GIT_PATH:
+    import os
+    os.environ['GIT_PYTHON_GIT_EXECUTABLE'] = \
+        os.path.join(GitRunner._GIT_PATH, 'git')
+
+from .config import ConfigManager
+cfg = ConfigManager()
+
+from .log import lgr
+from datalad.utils import on_windows, get_encoding_info, get_envvars_info
+
+lgr.log(5, "Instantiating ssh manager")
+from .support.sshconnector import SSHManager
+ssh_manager = SSHManager()
+atexit.register(ssh_manager.close, allow_fail=False)
 
 atexit.register(lgr.log, 5, "Exiting")
 

--- a/datalad/__init__.py
+++ b/datalad/__init__.py
@@ -28,12 +28,9 @@ try:
     colorama.init()
     atexit.register(colorama.deinit)
 except ImportError as e:
-    if on_windows:
-        from datalad.dochelpers import exc_str
-        lgr.warning(
-            "'colorama' Python module missing, terminal output may look garbled [%s]",
-            exc_str(e))
-    pass
+    # To not interfer with carefully crafted order of imports
+    # delay possibly issuing a warning until all needed imports are done
+    colorama = None
 
 # Other imports are interspersed with lgr.debug to ease troubleshooting startup
 # delays etc.
@@ -56,6 +53,14 @@ lgr.log(5, "Instantiating ssh manager")
 from .support.sshconnector import SSHManager
 ssh_manager = SSHManager()
 atexit.register(ssh_manager.close, allow_fail=False)
+
+if on_windows and colorama is None:
+    from datalad.dochelpers import exc_str
+
+    lgr.warning(
+        "'colorama' Python module missing, terminal output may look garbled ["
+        "%s]",
+        exc_str(e))
 
 atexit.register(lgr.log, 5, "Exiting")
 

--- a/datalad/__init__.py
+++ b/datalad/__init__.py
@@ -168,8 +168,12 @@ def teardown_package():
     lgr.debug("Printing versioning information collected so far")
     from datalad.support.external_versions import external_versions as ev
     print(ev.dumps(query=True))
-    print("Obscure filename: str=%s repr=%r"
-            % (OBSCURE_FILENAME.encode('utf-8'), OBSCURE_FILENAME))
+    try:
+        print("Obscure filename: str=%s repr=%r"
+                % (OBSCURE_FILENAME.encode('utf-8'), OBSCURE_FILENAME))
+    except UnicodeEncodeError as exc:
+        from .dochelpers import exc_str
+        print("Obscure filename failed to print: %s" % exc_str(exc))
     def print_dict(d):
         return " ".join("%s=%r" % v for v in d.items())
     print("Encodings: %s" % print_dict(get_encoding_info()))

--- a/datalad/cmdline/main.py
+++ b/datalad/cmdline/main.py
@@ -35,6 +35,7 @@ from datalad.support.exceptions import CommandError
 from .helpers import strip_arg_from_argv
 from ..utils import setup_exceptionhook, chpwd
 from ..utils import assure_unicode
+from ..utils import on_msys_tainted_paths
 from ..dochelpers import exc_str
 
 
@@ -450,10 +451,25 @@ def add_entrypoints_to_interface_groups(interface_groups):
             lgr.warning('Failed to load entrypoint %s: %s', ep.name, exc_str(e))
             continue
 
+def _fix_datalad_ri(s):
+    """Fixup argument if it was a DataLadRI and had leading / removed
+
+    See gh-2643
+    """
+    if s.startswith('//') and (len(s) == 2 or (len(s) > 2 and s[2] != '/')):
+        lgr.info(
+            "Changing %s back to /%s as it was probably changed by MINGW/MSYS, "
+            "see http://www.mingw.org/wiki/Posix_path_conversion", s, s)
+        return "/" + s
+    return s
+
 
 def main(args=None):
     lgr.log(5, "Starting main(%r)", args)
     args = args or sys.argv
+    if on_msys_tainted_paths:
+        # Possibly present DataLadRIs were stripped of a leading /
+        args = [_fix_datalad_ri(s) for s in args]
     # PYTHON_ARGCOMPLETE_OK
     parser = setup_parser(args)
     try:

--- a/datalad/cmdline/tests/test_main.py
+++ b/datalad/cmdline/tests/test_main.py
@@ -19,7 +19,11 @@ from six.moves import StringIO
 from mock import patch
 
 import datalad
-from ..main import main, fail_with_short_help
+from ..main import (
+    main,
+    fail_with_short_help,
+    _fix_datalad_ri,
+)
 from datalad import __version__
 from datalad.cmd import Runner
 from datalad.ui.utils import get_console_width
@@ -261,3 +265,13 @@ def test_fail_with_short_help():
                  "        mother\n"
                  "        father\n"
                  "Hint: You can become one\n")
+
+def test_fix_datalad_ri():
+    assert_equal(_fix_datalad_ri('/'), '/')
+    assert_equal(_fix_datalad_ri('/a/b'), '/a/b')
+    assert_equal(_fix_datalad_ri('//'), '///')
+    assert_equal(_fix_datalad_ri('///'), '///')
+    assert_equal(_fix_datalad_ri('//a'), '///a')
+    assert_equal(_fix_datalad_ri('///a'), '///a')
+    assert_equal(_fix_datalad_ri('//a/b'), '///a/b')
+    assert_equal(_fix_datalad_ri('///a/b'), '///a/b')

--- a/datalad/plugin/wtf.py
+++ b/datalad/plugin/wtf.py
@@ -13,10 +13,38 @@ __docformat__ = 'restructuredtext'
 from collections import OrderedDict
 from datalad.interface.base import Interface
 from datalad.interface.base import build_doc
+from datalad.utils import getpwd
 
 # wording to use for items which were considered sensitive and thus not shown
 _HIDDEN = "<Sensitive data, use -s=some or -s=all to show>"
 
+
+def get_max_path_length(top_path=None, maxl=1000):
+    """Deduce the maximal length of the filename in a given path
+    """
+    if not top_path:
+        top_path = getpwd()
+    import os
+    import random
+    from datalad import lgr
+    from datalad.dochelpers import exc_str
+    from datalad.support import path
+    prefix = path.join(top_path, "dl%d" % random.randint(1 ,100000))
+    # some smart folks could implement binary search for this
+    max_path_length = None
+    for i in range(maxl-len(prefix)):
+        filename = prefix + '_' * i
+        path_length = len(filename)
+        try:
+            with open(filename, 'w') as f:
+                max_path_length = path_length
+        except Exception as exc:
+            lgr.debug(
+                "Failed to create sample file for length %d. Last succeeded was %s. Exception: %s",
+                path_length, max_path_length, exc_str(exc))
+            break
+        os.unlink(filename)
+    return max_path_length
 
 @build_doc
 class WTF(Interface):
@@ -179,6 +207,12 @@ Metadata
                     cfg[k] = _HIDDEN
 
         from datalad.version import __version__, __full_version__
+
+        pwd = getpwd()
+        max_path_length = get_max_path_length(pwd)
+        max_path_msg = '%d  You are left with %d characters under %s' \
+                       % (max_path_length, max_path_length - (len(pwd) + 1), pwd) \
+            if max_path_length is not None else "UNKNOWN"
         text = report_template.format(
             datalad=_format_dict([
                 ('Version', __version__),
@@ -193,7 +227,9 @@ Metadata
                 ('Distribution',
                  ' '.join([_t2s(pl.dist()),
                            _t2s(pl.mac_ver()),
-                           _t2s(pl.win32_ver())]).rstrip())
+                           _t2s(pl.win32_ver())]).rstrip()),
+                ('Max path length',
+                 max_path_msg)
             ], indent=True),
             loc=_format_dict(get_encoding_info(), indent=True),  # , fmt="{}={!r}"),
             env=_format_dict(get_envvars_info(), fmt="{}={!r}"),

--- a/datalad/utils.py
+++ b/datalad/utils.py
@@ -1215,7 +1215,17 @@ def getpwd():
     If no PWD found in the env, output of getcwd() is returned
     """
     try:
-        return os.environ['PWD']
+        pwd = os.environ['PWD']
+        if on_windows and pwd and pwd.startswith('/'):
+            # It should be a path from MSYS.
+            # - it might start with a drive letter or not
+            # - it seems to be "illegal" to have a single letter directories
+            #   under / path, i.e. if created - they aren't found
+            # - 'ln -s' does not fail to create a "symlink" but it just copies!
+            #   so we are not likely to need original PWD purpose on those systems
+            # Verdict:
+            return os.getcwd()
+        return pwd
     except KeyError:
         return os.getcwd()
 

--- a/datalad/utils.py
+++ b/datalad/utils.py
@@ -66,6 +66,9 @@ platform_system = platform.system().lower()
 on_windows = platform_system == 'windows'
 on_osx = platform_system == 'darwin'
 on_linux = platform_system == 'linux'
+on_msys_tainted_paths = on_windows \
+                        and 'MSYS_NO_PATHCONV' not in os.environ \
+                        and os.environ.get('MSYSTEM', '')[:4] in ('MSYS', 'MING')
 try:
     linux_distribution_name, linux_distribution_release \
         = platform.linux_distribution()[:2]

